src/loop.jl

```diff
@@ -1,9 +1,25 @@
+"""Convert a dict with string keys to a NamedTuple, for python-eqsue kwargs splatting"""
+function symbolize(d::Dict{String,Any})::NamedTuple
+    return (; (Symbol(k) => v for (k,v) in d)...)
+end
+
+
+function new_model(factory_like)::Union{JuMP.Model, JuMP.MOI.AbstractOptimizer}
+    if isa(factory_like, Gurobi.Env)
+        return JuMP.direct_model(Gurobi.Optimizer(factory_like))
+    else
+        return JuMP.Model(factory_like)
+    end
+end
+
+
 """
-    interval_loop(env, model_kwargs, solver_kwargs, interval, n_interval,
+    interval_loop(factory_like, model_kwargs, solver_kwargs, interval, n_interval,
                   start_index, inputfolder, outputfolder)
 
 Given:
-- a Gurobi environment `env`
+- optimizer instantiation object `factory_like`:
```

Comment:
I'm wondering about the term `factory_like`...I'm not sure exactly what it communicates. Maybe something like `factory_spec` or `factory_env` or `optimization_factory_env`, etc?

Reply:
It's either an optimization factory _or_ a Gurobi environment.